app/components/post/post.js

```diff
@@ -485,176 +245,51 @@ class Post extends PureComponent {
-                    <MattermostIcon
-                        color={theme.centerChannelColor}
-                        height={PROFILE_PICTURE_SIZE}
-                        width={PROFILE_PICTURE_SIZE}
+        return (
+            <View style={[style.container, this.props.style, selected]}>
+                <View style={[style.profilePictureContainer, (isPostPendingOrFailed(post) && style.pendingPost)]}>
+                    <PostProfilePicture
+                        onViewUserProfile={this.viewUserProfile}
+                        postId={post.id}
                     />
                 </View>
-            );
-
-            displayName = (
-                <FormattedText
-                    id='post_info.system'
-                    defaultMessage='System'
-                    style={style.displayName}
-                />
-            );
-
-            messageStyle = [style.message, style.systemMessage];
-        } else if (post.props && post.props.from_webhook) {
-            if (config.EnablePostIconOverride === 'true') {
-                const icon = post.props.override_icon_url ? {uri: post.props.override_icon_url} : webhookIcon;
-                profilePicture = (
-                    <View style={style.profilePicture}>
-                        <Image
-                            source={icon}
-                            style={{
-                                height: PROFILE_PICTURE_SIZE,
-                                width: PROFILE_PICTURE_SIZE,
-                                borderRadius: PROFILE_PICTURE_SIZE / 2
-                            }}
+                <View style={style.messageContainerWithReplyBar}>
+                    {!commentedOnPost && this.renderReplyBar()}
+                    <View style={[style.rightColumn, (commentedOnPost && isLastReply && style.rightColumnPadding)]}>
+                        <PostHeader
+                            postId={post.id}
+                            commentedOnUserId={commentedOnPost && commentedOnPost.user_id}
+                            createAt={post.create_at}
+                            onPress={this.handlePress}
+                            onViewUserProfile={this.viewUserProfile}
+                            renderReplies={renderReplies}
+                            theme={theme}
+                        />
+                        <PostBody
+                            canDelete={this.state.canDelete}
+                            canEdit={this.state.canEdit}
+                            navigator={this.props.navigator}
+                            onFailedPostPress={this.handleFailedPostPress}
+                            onPostDelete={this.handlePostDelete}
+                            onPostEdit={this.handlePostEdit}
+                            onPress={this.handlePress}
+                            postId={post.id}
+                            renderReplyBar={!!commentedOnPost ? this.renderReplyBar : emptyFunction} //eslint-disable-line
```

Comment:
Why disable eslint on this line? You can either just treat `commentedOnPost` as falsey and go use `Boolean(commentedOnPost)`